src/compiler/crystal/syntax/parser.cr

```diff
@@ -4126,6 +4126,10 @@ module Crystal
         block = parse_block(block)
       end
 
+      if block && block_arg
+        raise "can't forward block argument and specify block together", location
```

Comment:
Do we call this "forwarding a block" elsewhere? Because that's not the only usecase, and the message might be slightly confusing for

```cr
foo(&-> { 1 }) { 2 }
```

Reply:
"Forward" comes from the Crystal book, I guess "can't use captured and non-captured blocks together" might also work.